File: javascript/matrix/matrix.js
class Matrix {
    constructor(str) {
        this.rows = [];
        this.columns = [];

        // fill the rows array
        for (var row_str of str.split("\n")) {
            var row = row_str.split(" ").map(function(t) {
                return parseInt(t);
            });
            this.rows.push(row);
        }

        // fill the columns array
        for (var y in this.rows) {
            for (var x in this.rows[y]) {
                // create empty colum array if undefined
                if (this.columns[x] == undefined) {
                    this.columns[x] = [];
                }
                // x, y <- y, x
                this.columns[x][y] = this.rows[y][x];
            }
        }
    }
}

module.exports = Matrix;

if (require.main === module) {
    new Matrix("0 1 2\n3 4 5");
}

File: python/matrix/matrix.py
from typing import List

class Matrix(object):
    def __init__(self, input: str):
        self.rows: List[List[int]] = [
            [int(i) for i in input_row.split(' ')]
            for input_row in input.split('\n')
        ]
        self.columns: List[List[int]] = [
            list(column)
            for column in zip(*self.rows)
        ]

    def row(self, index: int) -> List[int]:
        return self.rows[index-1]

    def column(self, index: int) -> List[int]:
        return self.columns[index-1]
